API/application/iotjs.py

```diff
@@ -121,10 +111,15 @@ def apply_patches(self):
         utils.execute(paths.IOTJS_JERRY_PATH, 'git', ['apply', jerry_memstat_patch])
         utils.execute(paths.IOTJS_JERRY_PATH, 'git', ['add', '-u'])
 
-    def build(self):
+    def build(self, device):
         '''
         Build IoT.js for the target device/OS and for Raspberry Pi 2.
         '''
+
+        # prebuild the OS
+        os = device.get_os()
+        os.prebuild(self)
```

Comment:
Are you sure that we need to prebuild the `os` in the app's build method? (I think they are two different things, so the app should not build the `os`, to be they independent.)

I think it would be clearer if the `os.prebuild` could happen in the device's `__init__` method (after the `os` instantiation), because the `os` is the part of the `device` right now.

Reply:
There are some function calls in the `os.prebuild` method which depends on the application. This is a bigger change compared to the previous state where these functions were called from the os module's `__init__` method. That solution was depend on both of the application and the device (which I think is worse).
But I'am agree with you, I will thinking on a proper solution to eliminate these dependencies.